Spelling

jsoref opened the pull request:
I'm going to actively ignore your submission guidelines.
As a general rule, whenever I submit large changesets to large projects, people insist on changes. As such, there's no point in me signing a submission at this time.
The current summary of each changeset represents a distinct word family misspelling. Sometimes projects have a preferred language family (en-US/en-GB being the most common), as such it's handy to be able to easily drop changesets. It's also not uncommon for some changes to need to be split out into special PRs (e.g. if they would constitute an API change).

**- What I did**
I spellchecked the docker/docker repository (and then dealt with the fact that it became moby/moby)

**- How I did it**
I used my own personal tool to generate a list of tokens not found in `/usr/share/dict/words` -- I then manually consider each remaining token. Typically I'll let Google Suggest suggest replacement words.

**- How to verify it**
Review the diff.

**- Description for the changelog**
86 word families
85 files changed, 148 insertions(+), 148 deletions(-)


**- A picture of a cute animal (not mandatory but encouraged)**
![image](https://cloud.githubusercontent.com/assets/2119212/26288409/c30602da-3e5f-11e7-87e4-f571c5de29a4.png)

jsoref committed:
Spelling fixes

* additional
* ambiguous
* anonymous
* anything
* application
* because
* before
* building
* capabilities
* circumstances
* commit
* committer
* compresses
* concatenated
* config
* container
* container's
* current
* definition
* delimiter
* disassociates
* discovery
* distributed
* doesnotexist
* downloads
* duplicates
* either
* enhancing
* enumerate
* escapable
* exactly
* expect
* expectations
* expected
* explicitly
* false
* filesystem
* following
* forbidden
* git with
* healthcheck
* ignore
* independent
* inheritance
* investigating
* irrelevant
* it
* logging
* looking
* membership
* mimic
* minimum
* modify
* mountpoint
* multiline
* notifier
* outputting
* outside
* overridden
* override
* parsable
* plugins
* precedence
* propagation
* provided
* provides
* registries
* repositories
* returning
* settings
* should
* signals
* someone
* something
* specifically
* successfully
* synchronize
* they've
* thinking
* uninitialized
* unintentionally
* unmarshaling
* unnamed
* unreferenced
* verify

Signed-off-by: Josh Soref <jsoref@gmail.com>
Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

GordonTheTurtle commented:
<!-- AUTOMATED:POULE:DCO-EXPLANATION -->
Please sign your commits following these rules:
https://github.com/moby/moby/blob/master/CONTRIBUTING.md#sign-your-work
The easiest way to do this is to amend the last commit:
~~~console
$ git clone -b "spelling" git@github.com:jsoref/moby.git somewhere
$ cd somewhere
$ git rebase -i HEAD~842354289856
editor opens
change each 'pick' to 'edit'
save the file and quit
$ git commit --amend -s --no-edit
$ git rebase --continue # and repeat the amend for each commit
$ git push -f
~~~

Amending updates the existing PR. You **DO NOT** need to open a new one.

jsoref commented:
@GordonTheTurtle: sorry. no. please read my PR description. I'm not going to spend time signing something until people talk to me about the actual content / what real changes they would like.

I'm willing to perform one round of signing at the very end, but no sooner.

duglin commented:
please squash down to one commit and sign that one. Generally, spelling changes are good and accepted. My quick scan of these looked pretty good for the most part, but the tests aren't happy so I suspect some of them you'll have to look at (and possibly revert if necessary), but if (as you said) some are API changes then put those in different PRs - keep this one limited to just spelling changes so it can go in easily.

jsoref commented:
dco-signed/janky both appear to be complaining about the signature;
windowsRS1 seems to be complaining about the fact that its networking stack is broken

fntlnz commented:
@jsoref  triggered the windows build a bunch of times but looks like some networking part is broken right now. Warned other maintainers so that someone with powers can fix it.

For the DCO you have to sign every commit, I would suggest you to squash your commits prior than doing so

jsoref commented:
Thanks. I'm not going to squash until someone tells me that the changes are acceptable. Otherwise, it's hard to tease out which other things need to be teased out.

If someone could poke me when one or the other of the builds passes (or do their own local build and tell me they're ok w/ things), then I'll see about squashing.

Also, if someone wants to propose a commit message, I'd appreciate it. I don't think that my normal commit message (spelling: work / spelling fixes + bulleted list of fixes) matches your style.

I don't mind working with projects to fit into things, but I'd rather provide patches than be mired in paperwork.

AkihiroSuda commented on api/types/plugins/logdriver/entry.pb.go:
```diff
@@ -428,7 +428,7 @@ func skipEntry(dAtA []byte) (n int, err error) {
 
 var (
-	ErrInvalidLengthEntry = fmt.Errorf("proto: negative length found during unmarshaling")
+	ErrInvalidLengthEntry = fmt.Errorf("proto: negative length found during unmarshalling")
```
Please do not edit this file, which is automatically generated from `entry.proto`.

AkihiroSuda commented on contrib/selinux-fedora-24/docker-engine-selinux/docker.te:
```diff
@@ -160,7 +160,7 @@ corenet_tcp_sendrecv_generic_node(docker_t)
 corenet_tcp_bind_all_ports(docker_t)
 corenet_tcp_connect_http_port(docker_t)
-corenet_tcp_connect_commplex_main_port(docker_t)
+corenet_tcp_connect_complex_main_port(docker_t)
```
commplex seems correct
http://oss.tresys.com/docs/refpolicy/api/kernel_corenetwork.html

AkihiroSuda commented on distribution/pull_v2_test.go:
```diff
@@ -129,7 +129,7 @@ func TestValidateManifest(t *testing.T) {
 	err = json.Unmarshal(goodManifestBytes, &goodSignedManifest)
 	if err != nil {
-		t.Fatal("error unmarshaling manifest:", err)
+		t.Fatal("error unmarshalling manifest:", err)
```
"marshaling" is widely accepted spelling

jsoref replied:
@AkihiroSuda : I only change a word like that if it's used inconsistently, in this case, here are the preexisting instances of the alternate spelling that I arbitrarily selected:
```
CHANGELOG.md:- Fix unmarshalling of Command and Entrypoint
CHANGELOG.md:* Use same error handling while unmarshalling CMD and ENTRYPOINT
registry/endpoint_v1.go:                logrus.Debugf("Error unmarshalling the _ping PingResult: %s", err)```

I'm happy to drop the current change, but I will replace it w/ a change to these three files to go the other direction.

jsoref commented:
@fntlnz, @AkihiroSuda: 3 successful, should I be able to collapse and sign things at this point?

fntlnz commented:
@jsoref the windows node has been rebooted, triggered a build

vdemeester reviewed (approved):
LGTM 👼

thaJeztah reviewed (changes requested):
changes LGTM, but please squash and sign off

fntlnz commented:
LGTM

jsoref commented:
Windows seems to have dns issues again

thaJeztah commented:
```
00:43:35 These files are not properly gofmt'd:
00:43:35  - api/types/filters/parse_test.go
00:43:35  - builder/dockerfile/buildargs_test.go
00:43:35 
00:43:35 Please reformat the above files using "gofmt -s -w" and commit the result.
```

looks like some files need a gofmt

jsoref commented:
@thaJeztah: that should not be my responsibility as the author of this commit.

There are at least two lines apparently considerably more out of compliance than my line:
```go
                        t.Fatalf("Expected false for %v on %v, got true", sources, args)
...
                        "created": map[string]bool{"anything": true, "to*": true}},
...
                        t.Fatalf("Expected true for %v on %v, got false", source, args)
```

As a general rule, for tests you should always (conceptually) run two:
1. against the current system
2. against the proposed system

If the proposed system does not result in more bugs than the previous but does result in fewer, then you should accept it.

Anyone who sets the bar beyond that for spelling fixes is doing a great disservice to everyone.
I certainly wouldn't have any incentive to contribute to your project.

I use spelling fixes to test the waters. Your project is failing miserably. Your baseline coverage is clearly faulty in a number of ways, your automated Windows tests are broken (and you don't have a hook to automatically recognize this -- you should respond to a failed build by trying a baseline w/o the feature to confirm that the box is functioning properly -- it clearly isn't).

If your code has other preexisting bugs, then it's your job as people involved in the project to recognize them and resolve them, not to force them on new contributors. Forcing new contributors to fix additional unrelated bugs is not a good way to welcome potential contributors.

To some extent, I may have access to additional development resources that I could be able to assign to work on moby/docker. Given the pushback and time wasted trying to contribute trivial fixes, I'm less inclined to do so.

vdemeester commented:
> @thaJeztah: that should not be my responsibility as the author of this commit.

@jsoref actually it is. The fact that code should be properly formatted is even written in our [Contributing guidelines](https://github.com/moby/moby/blob/master/CONTRIBUTING.md#conventions). This is an automated check, that *checks* that file are properly formatted (using the `gofmt` tools which is in the built-in tools of golang). It is the same for **any** pull request.

> There are at least two lines apparently considerably more out of compliance than my line:

It's not about who's line are in fault or not, it's just about are those files formatted using `gofmt` or not. If `gofmt` is happy with those lines, we are that's all (just in terms of formatting).

> I use spelling fixes to test the waters. Your project is failing miserably. Your baseline coverage is clearly faulty in a number of ways, your automated Windows tests are broken (and you don't have a hook to automatically recognize this -- you should respond to a failed build by trying a baseline w/o the feature to confirm that the box is functioning properly -- it clearly isn't).

Those are completely and orthogonal issues. First, the fact that automated windows have problems is known and that we have flakey integration test in general too. We are working on it.

> If your code has other preexisting bugs, then it's your job as people involved in the project to recognize them and resolve them, not to force them on new contributors. Forcing new contributors to fix additional unrelated bugs is not a good way to welcome potential contributors.

We are not, in any way, asking you to fix any of those. @thaJeztah is just pointing out that the build fails on the `gofmt` step which is, as described in the contributing file, a requirement. *And, yeah, I don't know any code that doesn't have bugs 👼*.

We welcome PRs very much, and we value a lot your work on those spelling issues and the time you invested in it. We only asked that the main requirements for a PR (i.e. sign-off, code formatted, anything described in `CONTRIBUTING.md` really) is there, nothing more, nothing less (the flaky failure on the tests we will handle them and restart the jobs that failed 😉).

jsoref commented:
@vdemeester: Ring me when you've fixed your base so that the file passes that check. Or drop that change. I don't care. It's totally inappropriate for this PR for me to reformat files. Anyone who thinks it's appropriate is crazy.

Simply put, if I made a million changes to your codebase, refactored files, reindented files, fixed spelling, fixed a dozen other bugs, and made it a single PR, you'd be out of your mind to accept that PR. Instead, you'd ask me to split it up. And if you actually accepted it, I'd never consider using your software for anything.

If your system was more welcoming, I could be induced into writing a PR to run `gofmt` across your entire codebase. But in case it isn't remotely obvious, your system is not close to that, and I certainly won't.

vdemeester commented on builder/dockerfile/buildargs_test.go:
```diff
@@ -37,7 +37,7 @@ func TestGetAllAllowed(t *testing.T) {
 		"ArgWithDefaultInDockerfile":          "fromdockerfile1",
 		"ArgNoDefaultInDockerfileFromOptions": "fromopt3",
 		"ArgFromMeta":                         "frommeta1",
-		"ArgFromMetaOverriden":                "fromdockerfile3",
```
This changes make the `gofmt` fails, because **now** this `"fromdockerfile3"` is mis-aligned with the others.

jsoref replied:
I'm happy to fix this.

jsoref replied:
(I thought I pushed an update with this, I guess it'll have to wait until evening)

vdemeester commented on api/types/filters/parse_test.go:
```diff
@@ -228,7 +228,7 @@ func TestArgsMatch(t *testing.T) {
 			"created": map[string]bool{"tod": true}},
 		}: "created",
 		&Args{map[string]map[string]bool{
-			"created": map[string]bool{"anyting": true, "to*": true}},
```
This file didn't pass `gofmt` before for some reason…

jsoref replied:
This is the only one that bothers me.

vdemeester reviewed (commented):
So, one file didn't pass the `gofmt` test before, not sure why (might be an old one that was added *before* we added the `gofmt` checks). Errors are human too 

> I don't care. It's totally inappropriate for this PR for me to reformat files. Anyone who thinks it's appropriate is crazy.

So what you are saying is that, it is totally inappropriate to follow contributing guidelines of a project ? Sometimes in open-source world, you have to fix some mistakes of the others, most of the times these are small. I really feel like running *`gofmt` on 2 files* is not such a big deal.

> Simply put, if I made a million changes to your codebase, refactored files, reindented files, fixed spelling, fixed a dozen other bugs, and made it a single PR, you'd be out of your mind to accept that PR. Instead, you'd ask me to split it up. And if you actually accepted it, I'd never consider using your software for anything.

I'm not sure I follow what you are trying to say…

Anyway, I have two proposition, tell me which one you're ok with.

- I submit a PR to fix the `gofmt` for this specific file, i.e. `parse_test.go` (and then you rebase on top of it)
- *or* I carry this commit, fixing the `gofmt` in your commit so you get the credit you deserve on fixing all those spelling issue.

I really am trying my best to get this merged in, I wish you will too  :pray:

jsoref commented:
I'd prefer the former over the latter. It's a purity thing. That said, it's your project. You can do what you like.

duglin commented:
I believe what happened is that no one else has touched `api/types/filters/parse_test.go` since we upgraded to a newer version of golang, and there are things in there (not on that one line he touched) that gofmt doesn't like.  When I run it I see lots of diff like this:
```
-		&Args{map[string]map[string]bool{
-			"created": map[string]bool{"tom": true}},
+		{map[string]map[string]bool{
+			"created": {"tom": true}},
```
And since we only check touched files, that's why all other PRs, and master, are ok.

jsoref commented:
@duglin: that makes sense. I hope you'll all agree that this is something that should be fixed by applying something (`gofmt -r -s -w`?) to the entire codebase. -- Also, you probably should change your system so that you hook up `gofmt -r -s -w` to some other part of things, either a periodic check, a process check when you change your go version, a thing you do before/after a release cycle, etc.

Note: I'm on `ppc64le`, and don't have `go` handy. (I did waste a little time trying to find a way to build it, but the obvious options didn't work, and I'm certainly not going to cross compile go just to get gofmt.)

duglin commented:
is `gofmt` not part of https://hub.docker.com/r/ppc64le/golang/ ?

jsoref commented:
Oh, I should have underlined that point. Building **Go** just to run `gofmt` on two files is way more work than I should have to do. Way more. The barrier to entry for committing spelling fixes should be a basic text editor.

@duglin: Dunno. I don't have docker on the computer I'm using. And I'm not `root` on the system.

tophj-ibm commented:
@jsoref just an FYI, go binaries are available for download on ppc64le as of 1.7.4 https://golang.org/dl/

@dug gofmt is part of ppc64le/golang, which is just a debian:jessie image with go already installed.

stevvooe commented:
@jsoref Were you able to update this with properly formatted code?

vieux commented:
any update @jsoref ?

jsoref commented:
That's being addressed in a different PR...

thaJeztah commented:
gofmt is done in this PR; https://github.com/moby/moby/pull/33405; I can rebase after that is merged, given that @jsoref commented that he didn't have a working environment at this moment (https://github.com/moby/moby/pull/33405#issuecomment-311938546)

thaJeztah commented:
rebased

AkihiroSuda reviewed (approved):
LGTM

AkihiroSuda commented:
windows failure seems unrelated
```
08:00:32 ----------------------------------------------------------------------
08:00:32 FAIL: docker_cli_restart_test.go:265: DockerSuite.TestRestartContainerwithRestartPolicy
08:00:32 
08:00:32 docker_cli_restart_test.go:282:
08:00:32     // Make sure we can stop/start (regression test from a705e166cf3bcca62543150c2b3f9bfeae45ecfa)
08:00:32     dockerCmd(c, "stop", id1)
08:00:32 c:/gopath/src/github.com/docker/docker/pkg/testutil/cmd/command.go:64:
08:00:32     t.Fatalf("at %s:%d - %s", filepath.Base(file), line, err.Error())
08:00:32 ... Error: at cli.go:48 - 
08:00:32 Command:  d:\CI\CI-aae806393\binary\docker.exe stop 131fc8f81970cd4a24d5a9933de351ec4f24201bb54f23a88c7256d617666b43
08:00:32 ExitCode: 1
08:00:32 Error:    exit status 1
08:00:32 Stdout:   
08:00:32 Stderr:   Error response from daemon: Cannot stop container 131fc8f81970cd4a24d5a9933de351ec4f24201bb54f23a88c7256d617666b43: Cannot kill container 131fc8f81970cd4a24d5a9933de351ec4f24201bb54f23a88c7256d617666b43: invalid container: 131fc8f81970cd4a24d5a9933de351ec4f24201bb54f23a88c7256d617666b43
08:00:32 
08:00:32 
08:00:32 Failures:
08:00:32 ExitCode was 1 expected 0
08:00:32 Expected no error
08:00:32 
08:00:32 
08:00:32 
08:00:32 ----------------------------------------------------------------------
```

trying once more

jsoref commented:
Thanks @thaJeztah

thaJeztah commented:
You're welcome!
